src/test/java/com/rocketden/main/service/RoomServiceTests.java

```diff
@@ -346,4 +348,49 @@ public void updateRoomSettingsNoRoomFound() {
                 roomService.updateRoomSettings("999999", noRoomRequest));
         assertEquals(RoomError.NOT_FOUND, exception.getError());
     }
+
+    @Test
+    public void removeUserSuccess() {
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+
+        room.setHost(host);
+        room.addUser(host);
+
+        User user = new User();
+        user.setNickname(NICKNAME_2);
+        user.setUserId(USER_ID_2);
+        room.addUser(user);
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(eq(ROOM_ID));
+
+        RoomDto response = roomService.removeUser(ROOM_ID, USER_ID_2);
+
+        verify(socketService).sendSocketUpdate(eq(response));
+        assertEquals(1, response.getUsers().size());
+        assertFalse(response.getUsers().contains(UserMapper.toDto(user)));
+    }
+
+    @Test
+    public void removeNonExistentUser() {
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+
+        room.setHost(host);
+        room.addUser(host);
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(eq(ROOM_ID));
+
+        ApiException exception = assertThrows(ApiException.class, () ->
+                roomService.removeUser(ROOM_ID, USER_ID_2));
+        assertEquals(UserError.NOT_FOUND, exception.getError());
+    }
 }
```

Comment:
Add test for room not found as well (and invalid permissions once that's implemented)